AndroidAnnotations/androidannotations-core/androidannotations/src/main/java/org/androidannotations/internal/core/handler/AbstractFragmentByHandler.java

```diff
@@ -17,37 +17,50 @@
 
 import static com.helger.jcodemodel.JExpr.cast;
 import static com.helger.jcodemodel.JExpr.invoke;
-import static com.helger.jcodemodel.JExpr.ref;
+
+import java.lang.annotation.Annotation;
 
 import javax.lang.model.element.Element;
+import javax.lang.model.element.ExecutableElement;
 import javax.lang.model.element.TypeElement;
 import javax.lang.model.type.TypeMirror;
 
 import org.androidannotations.AndroidAnnotationsEnvironment;
 import org.androidannotations.ElementValidation;
+import org.androidannotations.handler.MethodInjectionHandler;
 import org.androidannotations.helper.CanonicalNameConstants;
+import org.androidannotations.helper.InjectHelper;
 import org.androidannotations.holder.EComponentWithViewSupportHolder;
 import org.androidannotations.holder.EFragmentHolder;
 
+import com.helger.jcodemodel.IJAssignmentTarget;
 import com.helger.jcodemodel.IJExpression;
 import com.helger.jcodemodel.JBlock;
 import com.helger.jcodemodel.JMethod;
 
+public abstract class AbstractFragmentByHandler extends CoreBaseAnnotationHandler<EComponentWithViewSupportHolder>implements MethodInjectionHandler<EComponentWithViewSupportHolder> {
 
-public abstract class AbstractFragmentByHandler extends CoreBaseAnnotationHandler<EComponentWithViewSupportHolder> {
+	private final InjectHelper<EComponentWithViewSupportHolder> injectHelper;
 
+	private Class<? extends Annotation> targetClass;
 	protected String findFragmentMethodName;
 
-	public AbstractFragmentByHandler(Class<?> targetClass, AndroidAnnotationsEnvironment environment, String findFragmentMethodName) {
+	public AbstractFragmentByHandler(Class<? extends Annotation> targetClass, AndroidAnnotationsEnvironment environment, String findFragmentMethodName) {
 		super(targetClass, environment);
+		this.targetClass = targetClass;
 		this.findFragmentMethodName = findFragmentMethodName;
+		injectHelper = new InjectHelper<>(validatorHelper, this);
 	}
 
 	@Override
 	protected void validate(Element element, ElementValidation validation) {
-		validatorHelper.enclosingElementHasEnhancedViewSupportAnnotation(element, validation);
+		injectHelper.validate(targetClass, element, validation);
 
-		validatorHelper.extendsFragment(element, validation);
+		Element param = element;
```

Comment:
Shouldn't this method receive the real param? I mean a new method in `MethodInjectionHandler`, which will receive the first method parameter if the annotation is on the method, the method parameter if the annotation is on the parameter, and the field is the annotation is on the field.
So it is transparent for the handler itself.